### ISSUE TITLE
ESM / CommonJS confusion with slate-react / is-hotkey

### ISSUE THREAD

**Issue by jaked** [pikapkg/cdn#142]
I'm trying to use `slate-react`, which uses `is-hotkey`. For some reason the transpilation of `slate-react` tries to use `is-hotkey` as a CommonJS module even though it is served as an ESM module (I think this is what's happening?). In `slate-react` I see:

```
import __commonjs_module0 from "/-/is-hotkey@v0.1.8-WC8NeBDJCln4Q6xwwyOf/dist=es2020,mode=imports/optimized/is-hotkey.js";
const {isKeyHotkey} = __commonjs_module0
```

but the transpilation of `is-hotkey` is an ESM module, with a `default` that is not an object containing all the exports, so `isKeyHotkey` is undefined.

**Comment by drwpow** [comment]
You’re correct—that seems to be transformed incorrectly from `slate-react`. There is no `isKeyHotkey` exported on `default` as you stated. Though the issue is partially because of CommonJS <> ESM conversion, we still should be handling this better. I’ll look into where the transformation bug is.

> For some reason the transpilation of `slate-react` tries to use `is-hotkey` as a CommonJS module even though it is served as an ESM module

This would be one way to fix it—by getting `is-hotkey` to expose ESM. This library does publish ESM to npm at `src/index.js`. However, in its `package.json`, there’s nothing pointing to that `src/index.js` file (no `"module"` entrypoint, or `"exports"` map), so when `slate-react` imports it the CommonJS will be used instead. If `slate-react` had imported `is-hostkey/src/index.js` instead, then we would be serving ESM. But without something in `is-hotkey`’s package.json, there’s no way we would ever know that `src/index.js` and `lib/index.js` are the same file.

**Comment by jaked** [comment]
ok, thank you for the quick reply and detailed explanation!

**Comment by FredKSchott** [comment]
Oh interesting, our CJS->ESM conversion (powered by Rollup) seems to understand that the package is ESM compiled to CJS and so doesn't return the entire module as the `default`, as it normally would for a CJS->ESM converted module.

I actually think it's better to turn that off, and treat is-hotkey like any other CJS package, since we have our own "is this CJS" detection that this breaks.
